lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -192,27 +196,47 @@ def parse(body)
         JSON.parse(body)
       end
 
-      def commit(action, parameters, options = {})
+      def commit(action, parameters, options = {}, method='post')
         url = url(action, parameters, options)
         post = post_data(action, parameters)
         begin
-          raw = ssl_post(url, post, headers(post, options))
+          if method == 'post'
+            raw = ssl_post(url, post, headers(post, options))
+          else
+            raw = ssl_get(url(action, parameters, options), headers(""))
```

Comment:
Why are we calling url() function again? The URL construction is done in L200

Reply:
True removing it.